src/main/groovy/com/marklogic/gradle/task/MarkLogicTask.groovy

```diff
@@ -44,21 +44,29 @@ class MarkLogicTask extends DefaultTask {
 		project.hasProperty("mlAdminPassword") ? project.property("mlAdminPassword") : project.property("mlPassword")
 	}
 
+	DatabaseClient newClient() {
+		newClient(null)
+    }
+
 	/**
-	 * If the "database" property is set, then the DatabaseClient that's returned will use the App-Services port
-	 * (defaults to 8000) to connect to the given database. Otherwise, the DatabaseClient will try to connect to the
-	 * REST API server defined by mlRestPort.
+	 * If the "database" property is set or is passed as a parameter, then the DatabaseClient that's returned will use
+	 * the App-Services port (defaults to 8000) to connect to the given database. Otherwise, the DatabaseClient will
+	 * try to connect to the REST API server defined by mlRestPort.
 	 * @return
 	 */
-	DatabaseClient newClient() {
+	DatabaseClient newClient(String database) {
 		if (project.hasProperty("database")) {
 			println "Connecting via the App-Services port to database: " + project.property("database")
 			return getAppConfig().newAppServicesDatabaseClient(project.property("database"))
 		}
+		else if (database != null){
```

Comment:
I think this should come first - if a client passes in a non-null value, use that. Else, check for the project property. Else, call newDatabaseClient.

Reply:
https://github.com/marklogic-community/ml-gradle/pull/416/commits/cabc11f03c4ef626b4e11eb2e7086b2c6b63ce87 changes the proirity of task property.